docs/notebooks/how_to_build_and_optimize_qaoa_using_swap_strategy.ipynb

```diff
@@ -0,0 +1,742 @@
+      "text/plain": [
+       "<Figure size 2210.55x2709 with 1 Axes>"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from qiskit.circuit.library import QAOAAnsatz\n",
+    "from qiskit.quantum_info import SparsePauliOp\n",
+    "\n",
+    "cost_operator = SparsePauliOp.from_list(cost_hamiltonian)\n",
+    "ansatz = QAOAAnsatz(\n",
+    "    cost_operator,\n",
+    "    reps=2,\n",
+    ").decompose()\n",
```

Comment:
Can we pass some arguments to decompose to avoid decomposing the Hadamards at the beginning?

Reply:
I couldn't find the way to avoid decomposing the Hadamards...